File: src/RNetPi.Core/Packets/PacketS2CUpdateAvailable.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x7D
/// Update Available
/// Notifies the client that there's an update available
/// Data:
///     (String) Update Version
/// </summary>
public class PacketS2CUpdateAvailable : PacketS2C
{
    public const byte ID = 0x7D;

    public PacketS2CUpdateAvailable(string version)
    {
        WriteNullTerminatedString(version);
    }

    public override byte GetID() => ID;
}

File: src_js/server/packets/PacketS2CUpdateAvailable.js
const PacketS2C = require("./PacketS2C");

/**
 * Server -> Client
 * ID = 0x7D
 * Update Available
 * Notifies the client that theres an update available
 * Data:
 *      (Unsigned Char) Update Version
 */
class PacketS2CUpdateAvailable extends PacketS2C {
    constructor(version) {
        super();
        this._buffer.writeStringNT(version);
    }

    getID() {
        return 0x7D;
    }
}

module.exports = PacketS2CUpdateAvailable;
